internal/resources/kubernetessecret/resource_secret_test.go

```diff
@@ -45,23 +50,66 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
 		SecretResourceVar:  secretResourceVar,
 		SecretResourceName: fmt.Sprintf("%s.%s", ResourceName, secretResourceVar),
 		SecretName:         acctest.RandomWithPrefix("tf-sc-test"),
+		NamespaceName:      "default",
+		ClusterName:        acctest.RandomWithPrefix("tf-cluster"),
 	}
 }
 
-func TestAcceptanceForSecretResource(t *testing.T) {
-	clusterName := acctest.RandomWithPrefix("tf-cluster")
+func getConfigureContextFunc() func(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	if _, found := os.LookupEnv("ENABLE_SECRET_ENV_TEST"); !found {
+		return authctx.ProviderConfigureContextWithDefaultTransportForTesting
+	}
+
+	return authctx.ProviderConfigureContext
+}
+
+func getSetupConfig(config *authctx.TanzuContext) error {
+	if _, found := os.LookupEnv("ENABLE_SECRET_ENV_TEST"); !found {
+		return config.SetupWithDefaultTransportForTesting()
+	}
+
+	return config.Setup()
+}
 
+func TestAcceptanceForSecretResource(t *testing.T) {
 	testConfig := testGetDefaultAcceptanceConfig(t)
 
+	// If the flag to execute kubernetes secret tests is not found, run this as a unit test by setting up an http intercept for each endpoint
```

Comment:
Do you mean " run this as a **mock** test by setting up an http intercept for each endpoint" ?

Reply:
updated PTAl

Reply:
updated the comment